From 8bb2a0c02b305f8ae8611e501fe7dd3d2b4468a6 Mon Sep 17 00:00:00 2001
From: Joerg Bornemann <joerg.bornemann@qt.io>
Date: Tue, 11 Jun 2024 10:47:18 +0200
Subject: [PATCH] CMake: Re-enable lupdate/lconvert/lrelease in no-gui builds

This reverts 8dba0e48a0f7d3487b318a74f80f2d8e59c320f9 which disabled the
'linguist' feature if the 'printsupport' feature wasn't available.
However, the 'linguist' feature controls not only the Qt Linguist
application but also the command line tools lupdate, lconvert, and
lrelease. In no-gui builds, which also disable printsupport, the command
line tools were unexpectedly missing.

Fix the issue by extending the feature condition in
src/linguist/CMakeLists.txt. As drive-by, fix the FEATURE_png condition
that was still in QMake form from the initial conversion.

Fixes: QTBUG-126189
Task-number: QTBUG-125066
Change-Id: I59ebb82fd5823165b307ffbc967d7fd89a071ede
Reviewed-by:  Alexey Edelev <alexey.edelev@qt.io>
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit 4be1823e4d459c89717e791ef27fd463ad04cb2b)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
(cherry picked from commit aa9f8db49db2e7734c187445b8c3c56768f6e546)
---

--- a/configure.cmake
+++ b/configure.cmake
@@ -74,7 +74,6 @@ qt_feature("kmap2qmap" PRIVATE
 qt_feature("linguist" PRIVATE
     LABEL "Qt Linguist"
     PURPOSE "Qt Linguist can be used by translator to translate text in Qt applications."
-    CONDITION TARGET Qt::PrintSupport
 )
 qt_feature("pixeltool" PRIVATE
     LABEL "pixeltool"
--- a/src/linguist/CMakeLists.txt
+++ b/src/linguist/CMakeLists.txt
@@ -14,7 +14,8 @@ add_subdirectory(lrelease)
 add_subdirectory(lrelease-pro)
 add_subdirectory(lupdate)
 add_subdirectory(lupdate-pro)
-if(QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton AND TARGET Qt::Widgets AND NOT no-png)
+if(QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton
+    AND QT_FEATURE_png AND QT_FEATURE_printsupport AND TARGET Qt::Widgets)
     add_subdirectory(linguist)
 endif()
 
